src/lib/storage/ConnectionRecord.ts

```diff
@@ -0,0 +1,38 @@
+import { BaseRecord, RecordType } from './BaseRecord';
+import { ConnectionProps } from '../protocols/connections/domain/Connection';
+import { DidDoc } from '../protocols/connections/domain/DidDoc';
+import { InvitationDetails } from '../protocols/connections/domain/InvitationDetails';
+import { ConnectionState } from '../protocols/connections/domain/ConnectionState';
+import uuid from 'uuid/v4';
+
+export interface ConnectionStorageProps extends ConnectionProps {
+  id?: string;
+  tags: { [keys: string]: string };
+}
+
+export class ConnectionRecord extends BaseRecord implements ConnectionStorageProps {
```

Comment:
What if we try to use something like base `Entity` class and `props` attribute: https://github.com/stemmlerjs/white-label/blob/master/src/core/domain/Entity.ts

And use it in this way: https://github.com/stemmlerjs/white-label/blob/master/src/modules/vinyl/domain/album.ts

We would get rid off repeating lists of the same attributes in class and constructor.

I feel there is a lot extending and interfacing and not sure if it's necessary.

Reply:
> We would get rid off repeating lists of the same attributes in class and constructor.

If that's the only objective then `Object.assign(this, props)` should suffice but then we'd have typescript complain since Object.assign isn't type safe... Maybe we can use some suggestion from [here](https://stackoverflow.com/a/55583832)

Reply:
I don't like `Object.assign(...)` because that type problem you mentioned, neither the solution in given link 😄 We don't have to address this in this PR ;)